docs/readme-sync/v4.0/deploy-as-a-microservice/030 - use-optimizely-agent/010 - evaluate-rest-apis.md

```diff
@@ -0,0 +1,60 @@
+---
+title: "Evaluate REST APIs"
+excerpt: ""
+slug: "evaluate-rest-apis"
+hidden: false
+metadata: 
+  title: "Evaluate REST APIs - Optimizely Full Stack"
+createdAt: "2020-02-21T17:44:53.019Z"
+updatedAt: "2020-04-13T23:02:34.056Z"
+---
+Below is an example demonstrating the APIs capabilities. For brevity, we've chosen to illustrate the API usage with Python. Note that the API documentation is defined via an OpenAPI (Swagger) spec and can be viewed [here](https://library.optimizely.com/docs/api/agent/v1/index.htm).
+
+## Start an http session
+Each request made into Optimizely Agent is in the context of an Optimizely SDK Key. SDK Keys map API requests to a specific Optimizely Project and Environment. We can setup a global request header by using the `requests.Session` object.
+
+
+```python
+import requests
+
+s = requests.Session()
+s.headers.update({'X-Optimizely-SDK-Key': 'YOUR-SDK-KEY'})
+```
+The following examples will assume this session is being maintained.
+
+## Get current environment configuration
+The `/v1/config` endpoint returns a manifest of the current working environment.
+
+```python
+resp = s.get('http://localhost:8080/v1/config')
+env = resp.json()
+
+for key in env['featuresMap']:
+    print(key)
+```
+
+## Run a feature flag rule
+
+The Decide [endpoint](https://library.optimizely.com/docs/api/agent/v1/index.html#operation/decide) buckets a user into a feature flag variation (choosing between multiple enabled variations or one disabled variation) as part of a flag rule. Flag rules let you:
+- experiment using A/B tests
+- roll out feature flags progressively to a selected audience using targeted feature flag deliveries. 
+
+To run a flag rule, use
+
+```python
+# decide 1 flag. 
+params = { "keys": "my-feature-flag" }
+payload = { "userId": "test-user" }
```

Comment:
we can also send `userAttributes` in the payload.

Reply:
updated to
```python
payload = {
    "userId": "test-user",
    "userAttributes": {
        "attr1": "sample-attribute-1",
        "attr2": "sample-attribute-2"
    }
}
```